### PR TITLE
Using System Allocator to manage instances of `AssetDataStreamPrivate`.

### PR DESCRIPTION
**Pull request by Jackie9527**
Signed-off-by: Jackie9527 <80555200+Jackie9527@users.noreply.github.com>

## What does this PR do?

Using System Allocator to manage instances of `AssetDataStreamPrivate`.
## How was this PR tested?

Start `Editor` and open `Sponza`.

### CODE REVIEW

File: Code/Framework/AzCore/AzCore/Asset/AssetDataStream.cpp [hunk 1]
```diff
@@ -22,6 +22,7 @@ namespace AZ::Data
     {
         struct AssetDataStreamPrivate
         {
+            AZ_CLASS_ALLOCATOR(AssetDataStreamPrivate, SystemAllocator, 0);
```
Comment (mbalfour-amzn) [inline Code/Framework/AzCore/AzCore/Asset/AssetDataStream.cpp:25]: Nit: I think that last parameter is no longer needed, I'm pretty sure it got deprecated a few weeks ago. Ideally, new instances of this macro getting added shouldn't use it.

**Review by cgalvan — APPROVED** [review]

File: Code/Framework/AzCore/AzCore/Asset/AssetDataStream.cpp (hunk 1 above)
Comment (Jackie9527) [reply Code/Framework/AzCore/AzCore/Asset/AssetDataStream.cpp:25]: Maybe I could create another PR to remove the redundant third parameter referenced in all of code files.

Comment (Jackie9527) [reply Code/Framework/AzCore/AzCore/Asset/AssetDataStream.cpp:25]: I've submitted #14530, please help review code.

Comment (Jackie9527) [reply Code/Framework/AzCore/AzCore/Asset/AssetDataStream.cpp]: > Nit: I think that last parameter is no longer needed, I'm pretty sure it got deprecated a few weeks ago. Ideally, new instances of this macro getting added shouldn't use it.

done.
The last parameter has been removed.

**Review by cgalvan — APPROVED** [review]

**Pull request merged**